pkg/overrides/api.go

```diff
@@ -0,0 +1,179 @@
+package overrides
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/log/level"
+	"gopkg.in/yaml.v3"
+
+	"github.com/cortexproject/cortex/pkg/tenant"
+)
+
+const (
+	// HTTP status codes
+	StatusOK                  = 200
+	StatusBadRequest          = 400
+	StatusUnauthorized        = 401
+	StatusInternalServerError = 500
+
+	// Error messages
+	ErrInvalidJSON = "Invalid JSON"
+
+	// Runtime config errors
+	ErrRuntimeConfig = "runtime config read error"
+)
+
+type RuntimeConfigFile struct {
+	Overrides     map[string]map[string]interface{} `yaml:"overrides"`
+	HardOverrides map[string]map[string]interface{} `yaml:"hard_overrides"`
+}
```

Comment:
```suggestion
```

This overrides the other values saved in the runtime like ingester_limits

Use https://github.com/cortexproject/cortex/blob/2c07f003db2d95d7df94341260cab4de3799a81f/pkg/cortex/runtime_config.go#L26 instead

And make sure the API doesn't delete ingester_limits values and similar